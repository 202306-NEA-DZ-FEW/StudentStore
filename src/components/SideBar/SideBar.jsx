import React, { useEffect, useState } from "react";
import Link from "next/link";
import firebase from "firebase/app";
import "firebase/firestore";
import { collection, doc, getDoc } from "firebase/firestore";
import { db } from "../../util/firebase.js";

import { BiLogOut } from "react-icons/bi";

const SideBar = () => {
    const [open, setOpen] = useState(false);

    const [selectedLink, setSelectedLink] = useState(null);

    const [userInfo, setUserInfo] = useState(null);

    useEffect(() => {
        const fetchUserInfo = async () => {
            const userId = "e43JDIG05abGPsH43xEKBNsD49e2";
            const userInfoRef = doc(
                db,
                "userinfo",
                "e43JDIG05abGPsH43xEKBNsD49e2"
            );
            const userInfoSnapshot = await getDoc(userInfoRef);

            if (userInfoSnapshot.exists()) {
                const userInfoData = userInfoSnapshot.data();
                setUserInfo(userInfoData);
            }
        };
        fetchUserInfo();
    }, []);

    return (
        <div className='bg-gray-200 w-64 min-h-screen p-4 flex flex-col'>
            {userInfo && (
                <>
                    <div className='mb-4'>
                        <img
                            src={userInfo.photo}
                            alt='User'
                            className='w-16 h-16 rounded-full mb-2'
                        />
                        <h2 className='text-xl font-bold'>{`${userInfo.name} ${userInfo.surname}`}</h2>
                        <p className='text-gray-600'>{userInfo.email}</p>
                    </div>
                    <div className='mb-4'>
                        <p>{` ${userInfo.address.city}, ${userInfo.address.country}`}</p>
                    </div>
                    <div className='flex flex-col'>
                        <a href='#' className='text-blue-500 mb-2'>
                            Edit Profile
                        </a>
                        <a href='#' className='text-blue-500 mb-2'>
                            My Listings
                        </a>
                        <a href='#' className='text-blue-500 mb-2'>
                            My Orders
                        </a>
                    </div>
                </>
            )}
        </div>
    );
};

export default SideBar;
