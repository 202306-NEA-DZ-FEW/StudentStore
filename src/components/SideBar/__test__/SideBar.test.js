import renderer from "react-test-renderer";

import SideBar from "../SideBar";
jest.mock("../../../util/firebase", () => {
    return {
        initializeApp: jest.fn(),
    };
});
jest.mock("next/router", () => ({
    useRouter: () => ({
        pathname: "/",
    }),
}));

it("renders correctly", () => {
    const tree = renderer.create(<SideBar />).toJSON();
    expect(tree).toMatchSnapshot();
});
