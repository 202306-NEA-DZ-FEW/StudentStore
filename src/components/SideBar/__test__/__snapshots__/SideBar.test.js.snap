// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-gray-200 min-h-screen p-4 text-[#585785] flex flex-col w-64"
>
  <svg
    className="cursor-pointer text-4xl "
    fill="currentColor"
    height="1em"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M17.59 18L19 16.59 14.42 12 19 7.41 17.59 6l-6 6z"
    />
    <path
      d="M11 18l1.41-1.41L7.83 12l4.58-4.59L11 6l-6 6z"
    />
  </svg>
  <div
    className="flex flex-col items-center mt-20 mb-4 space-y-4"
  >
    <img
      alt="User"
      className="w-25 h-25 rounded-full mb-2"
      src=""
    />
  </div>
  <div
    className="flex flex-col items-center space-y-10"
  >
    <div
      className="flex items-center text-[#585785] text-2xl cursor-pointer font-semibold p-3 rounded-lg "
      onClick={[Function]}
    >
      <span
        className="mr-2"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 20h9"
          />
          <path
            d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
          />
        </svg>
      </span>
      <span
        className="hidden sm:inline"
      >
        Edit Profile
      </span>
    </div>
    <div
      className="flex items-center text-[#585785] text-2xl cursor-pointer font-semibold p-3 rounded-lg "
      onClick={[Function]}
    >
      <span
        className="mr-2"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.5 0a.5.5 0 0 1 .5.5.5.5 0 0 0 .5.5.5.5 0 0 1 .5.5V2a.5.5 0 0 1-.5.5h-5A.5.5 0 0 1 5 2v-.5a.5.5 0 0 1 .5-.5.5.5 0 0 0 .5-.5.5.5 0 0 1 .5-.5h3Z"
          />
          <path
            d="M3.5 1h.585A1.498 1.498 0 0 0 4 1.5V2a1.5 1.5 0 0 0 1.5 1.5h5A1.5 1.5 0 0 0 12 2v-.5c0-.175-.03-.344-.085-.5h.585A1.5 1.5 0 0 1 14 2.5v12a1.5 1.5 0 0 1-1.5 1.5h-9A1.5 1.5 0 0 1 2 14.5v-12A1.5 1.5 0 0 1 3.5 1Z"
          />
        </svg>
      </span>
      <span
        className="hidden sm:inline"
      >
        My Listings
      </span>
    </div>
    <div
      className="flex items-center text-[#585785] text-2xl cursor-pointer font-semibold  "
      onClick={[Function]}
    >
      <span
        className="mr-2"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.528 2.973a.75.75 0 0 1 .472.696v8.662a.75.75 0 0 1-.472.696l-7.25 2.9a.75.75 0 0 1-.557 0l-7.25-2.9A.75.75 0 0 1 0 12.331V3.669a.75.75 0 0 1 .471-.696L7.443.184l.01-.003.268-.108a.75.75 0 0 1 .558 0l.269.108.01.003 6.97 2.789ZM10.404 2 4.25 4.461 1.846 3.5 1 3.839v.4l6.5 2.6v7.922l.5.2.5-.2V6.84l6.5-2.6v-.4l-.846-.339L8 5.961 5.596 5l6.154-2.461L10.404 2Z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <span
        className="hidden sm:inline"
      >
        My Orders
      </span>
    </div>
  </div>
  <div
    className="mt-auto mb-4"
  >
    <div
      className="flex items-center text-red-500 text-lg cursor-pointer"
      onClick={[Function]}
    >
      <svg
        className="mr-2"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 13v-2H7V8l-5 4 5 4v-3z"
        />
        <path
          d="M20 3h-9c-1.103 0-2 .897-2 2v4h2V5h9v14h-9v-4H9v4c0 1.103.897 2 2 2h9c1.103 0 2-.897 2-2V5c0-1.103-.897-2-2-2z"
        />
      </svg>
      Logout
    </div>
  </div>
</div>
`;
